viam_cartographer.go

```diff
@@ -191,22 +199,36 @@ func New(
 	cancelCartoFacadeCtx, cancelCartoFacadeFunc := context.WithCancel(context.Background())
 
 	// use the override in testing if non nil
-	// otherwise use the lidar from deps as the
-	// timed sensor
-	timedLidar := testTimedSensorOverride
+	// otherwise use the sensors from deps as the
+	// timed sensors
+	timedLidar := testTimedLidarSensorOverride
+	timedIMU := testTimedIMUSensorOverride
 	if timedLidar == nil {
-		timedLidar = lidar
+		timedLidar = lidarObject
+	}
+	if timedIMU == nil {
+		timedIMU = imuObject
+	}
+
+	lidar := Lidar{
+		name:         lidarName,
+		dataRateMsec: lidarDataRateMsec,
+		actual:       lidarObject,
+		testing:      timedLidar,
```

Comment:
@nicksanford Are you ok with these naming conventions? Or do you have another preference

Reply:
Sorry, I don't really have context on what the goal of this change is. What is the goal?

Reply:
@nicksanford Goal is just to clean up the Cartographer Service struct by splitting out the lidar and IMU parameters into their own structs, such that one can call e.g. cfg.lidar.name to get the lidar name.

Reply:
I'm ok with the `actual` and `testing` naming conventions. 

I can't say that I fully understand how these new structs will work in the current state of this PR but those names specifically make sense to me.

Reply:
These new structs are to organize and reduce the complexity of the cartographerService struct. I felt that having 4 attributes all related to LiDAR was too confusing, particularly Lidar and LidarObject. These structs provide a more digestible grouping and naming conventions